### PR TITLE
discount code shows correct status and filter is fixed

### LINKED ISSUE #2557: Discount code status is incorrect
**Describe the bug**

Currently, the `Status` in `Discount Code` is incorrect. It retrieves data from server using a parameter `is_active` which is always `True` and so the status is always `Active`. I think it will be better to use `validTill` parameter of discount codes on frontend to dynamically calculate its status. In the screenshot attached that the validity of the discount code is expired, still it is showing active.
         Also, the filter is not working properly due to same reason.

**To Reproduce**
Steps to reproduce the behaviour:
1. Go to `Events/View/Tickets/`
2. Click on `Discount Codes`
3. See error

**Expected behaviour**

The discount code should show correct status, depending on `validTill` attribute

**Screenshots**

Screenshot_2019-04-10 Discount codes Tickets APK19 Events Open Event

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] Ubuntu 18.04
 - Browser [e.g. chrome, safari] Mozilla Firefox

**Additional context**

Working on it

### PR DESCRIPTION
**Pull request by shreyanshdwivedi**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Currently, the `Status` in `Discount Code` is incorrect. The status is always `Active`. I think it will be better to use validTill parameter of discount codes on frontend to dynamically calculate
Also, the filter is not working properly.

#### Changes proposed in this pull request:
- discount code shows correct status
- discount code status filter is fixed
- introduced expired status

Fixes #2557

### CODE REVIEW

File: app/models/discount-code.js [hunk 1]
```diff
@@ -25,12 +26,14 @@ export default ModelBase.extend({
...
-  isActive      : attr('boolean', { defaultValue: true }),
   createdAt     : attr('moment'),
 
-  tickets : hasMany('ticket'),
-  orders  : hasMany('order'),
-  event   : belongsTo('event', {
+  tickets  : hasMany('ticket'),
+  orders   : hasMany('order'),
+  isActive : computed('validTill', function() {
+    return (new Date() <= new Date(this.get('validTill')));
```
Comment (CosmicCoder96) [inline app/models/discount-code.js]: no -- you can't overwrite a property coming from the server - it's not the convention. Define a new propery called status. Calculate using both: isActive && <validTill condition> and then according to it decide the status of discount code. Make three statuses: active, inactive and expired. Active/inactive were created for the intention that an organizer can go at any time make a discount code inactive to prevents its usage without changing anything else. Valid till had nothing to do with it.

Comment (shreyanshdwivedi) [reply app/models/discount-code.js]: Ok. So what I plan to do is check if the <validTill condition> is satisfied or not. If it is satisfied, then `isActive` comes in play and the status is either `Active` or `Inactive` depending on `isActive`.
If <validTill condition> is not satisfied, then the status is marked as `Expired`. Is this fine?

Comment (CosmicCoder96) [reply app/models/discount-code.js]: yup @shreyanshdwivedi sounds good.

**Comment by shreyanshdwivedi** [comment]
@CosmicCoder96 I've made the required changes. Please review
Screenshot_2019-04-10 Discount codes Tickets APK19 Events Open Event(1)

**Comment by shreyanshdwivedi** [comment]
@pradeepgangwar @CosmicCoder96 please review

**Comment by pradeepgangwar** [comment]
@shreyanshdwivedi Just a little suggestion. Don't squash commits while PR is being reviewed as reviewers lose the track of the changes that you did after the review 😛 . We can squash it after the approval :)

**Comment by shreyanshdwivedi** [comment]
@pradeepgangwar sure, will keep that in mind for upcoming PRs :)

**Pull request merged**